lib/Dialect/Moore/MooreTypes.cpp

```diff
@@ -29,6 +29,16 @@ static void printMembers(AsmPrinter &printer,
 static ParseResult parseMooreType(AsmParser &parser, Type &type);
 static void printMooreType(Type type, AsmPrinter &printer);
 
+static std::optional<DenseMap<Attribute, Type>>
+getAllSubelementIndexMap(ArrayRef<StructLikeMember> members);
+static Type getTypeAtAllIndex(ArrayRef<StructLikeMember> members,
+                              Attribute index);
+
+static std::optional<DenseMap<Attribute, Type>>
+getAllSubelementIndexMap(ArrayRef<StructLikeMember> members);
+static Type getTypeAtAllIndex(ArrayRef<StructLikeMember> members,
+                              Attribute index);
+
```

Comment:
I don't think you need these forward declarations :slightly_smiling_face:

Reply:
Sorry, I should have removed it but forgot.